component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/RecordConvertersTest.java

```diff
@@ -344,6 +345,30 @@ void convertListObject(final JsonBuilderFactory jsonBuilderFactory, final JsonPr
         }
     }
 
+    @Test
+    void convertListVaryingObject(final JsonBuilderFactory jsonBuilderFactory, final JsonProvider jsonProvider,
+            final RecordBuilderFactory recordBuilderFactory, final RecordConverters converter) throws Exception {
+        try (final Jsonb jsonb = JsonbBuilder.create()) {
+            final Record record = converter
+                    .toRecord(new RecordConverters.MappingMetaRegistry(),
+                            Json
+                                    .createObjectBuilder()
+                                    .add("list", Json
+                                            .createArrayBuilder()
+                                            .add(Json.createObjectBuilder().add("name", "a").add("name1", "a1").build())
+                                            .add(Json.createObjectBuilder().add("name", "b").add("name2", "b2").build())
+                                            .build())
+                                    .build(),
+                            () -> jsonb, () -> new RecordBuilderFactoryImpl("test"));
+            final Collection<Record> list = record.getArray(Record.class, "list");
+            final Schema schema = record.getSchema().getEntries().get(0).getElementSchema();
+            // // FIXME: 7/9/21 : record -> schema -> entry(list) -> elementSchema != record -> list[x] -> schema
```

Comment:
if there's a potential fix, please create a jira issue and link it's preferable. thanks

Reply:
https://jira.talendforge.org/browse/TCOMP-1956